### DISCUSSION TITLE
部署合约到 testnet 或者 devnet 报错

### DISCUSSION THREAD

**Discussion by v1xingyue** [creatorsand/solana-co-learn#8] [Q&A] [answered]
不确定是不是网络问题，在config 设置 --url testnet 后，执行deploy 

显示第一行: 
![image](https://github.com/CreatorsDAO/all-in-one-solana/assets/974169/a14f2c3d-0c0f-45c0-b131-59c853ba0cfc)

然后，一直阻断，直到执行超时:

![image](https://github.com/CreatorsDAO/all-in-one-solana/assets/974169/abf3b873-71c1-4e22-aa04-299b5f64965d)

注: 该合约可以部署到本地。

**Comment by DaviRain-Su** [comment]
我感觉还是需要的sol不够

**Comment by v1xingyue** [comment, marked as answer]
可能存在几个原因：
1. rpc 节点太慢，可以更换快一点儿的 rpc node . 比如从 quicknode 新建一个
2. 账户sol 太少，可以多充一点儿，没有就airdrop 吧。 
3. 时间需要很长，需要拆分成好多个transaction 发送。 

![image](https://github.com/CreatorsDAO/all-in-one-solana/assets/974169/12158c81-0e83-45d3-b6d7-e9359ba7b009)

完成部署 😁 

![image](https://github.com/CreatorsDAO/all-in-one-solana/assets/974169/6dc6d0af-5da5-4166-9c80-2978498cafee)
